File: Codes from the AI-assistants/[Experimental] regex-redux/Specialized Prompt - TBR Mode/RegexRedux-gpt-specialized-tbr.cpp
std::string text = read_all(stdin);
std::vector<std::pair<std::regex, std::string>> rules = { … };
size_t total = 0;
bool changed = true;
while (changed) {
    changed = false;
    for (auto &r : rules) {
        std::smatch m;
        std::string out;
        size_t count = 0;
        std::string::const_iterator searchStart(text.cbegin());
        while (std::regex_search(searchStart, text.cend(), m, r.first)) {
            count++;
            out.append(searchStart, m[0].first);
            out.append(r.second);
            searchStart = m[0].second;
        }
        out.append(searchStart, text.cend());
        if (count) {
            total += count;
            text.swap(out);
            changed = true;
        }
    }
}
std::cout << total << "\n";

File: Codes from the AI-assistants/Old Experiments/[Experimental] regex-redux/Specialized Prompt - TBR Mode/RegexRedux-gpt-specialized-tbr.py
import sys, re
text = sys.stdin.read()
rules = [(re.compile(pat), rep) for pat,rep in patterns]
total = 0
changed = True
while changed:
    changed = False
    for regex, rep in rules:
        new, cnt = regex.subn(rep, text)
        if cnt > 0:
            total += cnt
            text = new
            changed = True
print(total)
